source/MaterialXCore/Node.cpp

```diff
@@ -76,8 +76,18 @@ NodeDefPtr Node::getNodeDef(const string& target, bool allowRoughMatch) const
     }
     vector<NodeDefPtr> nodeDefs = getDocument()->getMatchingNodeDefs(getQualifiedName(getCategory()));
     vector<NodeDefPtr> secondary = getDocument()->getMatchingNodeDefs(getCategory());
-    vector<NodeDefPtr> roughMatches;
     nodeDefs.insert(nodeDefs.end(), secondary.begin(), secondary.end());
+
+    // Recurse to data library if present.
+    if (getDocument()->hasDataLibrary())
```

Comment:
Is this additional code block still needed here, or will it now be handled by the data library checks within the two `getMatchingNodeDefs` calls above?

Reply:
Good catch, updated them.